File: app/src/main/java/com/go/sispentra/data/Transaksi.kt
package com.rw.keyboardlistener.com.go.sispentra.data

data class Transaksi(
    var id:Int,
    var type_transaksi:String,
    var nominal:Double,
    var status:String,
    var tgl_transaksi:String,
    var tgl_validasi_bendahara:String ? = null,
    var tgl_validasi_kolektor:String ? = null,
    var tgl_validasi_nasabah:String ? = null,
    var no_tabungan:String,
    var nasabah_name:String?,
    var kolektor_name:String?,
    var saldo:Int?=0
)

File: laravel/LPD_Android/app/Models/Transaksi.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Transaksi extends Model
{
    use HasFactory;
    protected $fillable = [
        'type_transaksi',
        'nasabah_id',
        'buku_tabungan_id',
        'tgl_transaksi',
        'tgl_validasi_bendahara',
        'tgl_validasi_kolektor',
        'tgl_validasi_nasabah',
        'nominal',
        'status'
    ];
    public function bukutabungan(){
        return $this->belongsTo(BukuTabungan::class,'buku_tabungan_id','id');
    }

    public function scopeFilter($query,array $filters)
    {
        $query->when($filters['type_transaksi'] ?? false,function($query,$type_transaksi){
            // return $query->where('fullname','like','%'.$fullname.'%');
            if($type_transaksi=="Semua"){
                return $query;
            }
            else {
                return $query->where('type_transaksi',$type_transaksi);
            }
        });
    }

}
